src/components/plots/ProductConcentrationPlot.tsx

```diff
@@ -125,7 +133,30 @@ const ProductConcentrationPlot: React.FC<ProductConcentrationPlotProps> = ({
                 color: AGENT_AB_COLOR,
             },
         },
+        shapes: [] as Partial<Plotly.Shape>[],
     };
+    if (dotsX.length > 0) {
+        const CIRCLE_RADIUS = 4;
+        const shapes = dotsX.map((x, i) => {
```

Comment:
Same here, can this be `shapes: Partial<Plotly.Layout["shapes"]>[]` or something?